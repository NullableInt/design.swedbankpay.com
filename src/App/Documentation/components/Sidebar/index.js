import React, { useEffect } from "react";

import { ComponentPreview, DocContainer, Property, Attribute, JavascriptDocs } from "@docutils";

const Sidebar = () => {
    useEffect(() => { console.log("init"); });

    return (
        <DocContainer docToc>
            <p className="lead">The sidebar is used to give users an easily available navigational bar at the left side of your web application.</p>
        </DocContainer>
    );
};

export default Sidebar;
