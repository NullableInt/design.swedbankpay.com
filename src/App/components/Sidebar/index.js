import React from "react";
import classnames from "classnames";

const Sidebar = ({ items, mode }) => {
    const sidebarClassname = classnames("sidebar",
        mode ? `sidebar-vertical-${mode}` : ""
    );

    return (
        <nav className={sidebarClassname}>
            <ul>
                {items.map(({ name, icon, subItems }, i) => (
                    <li key={`sidebar-item-${name}-${i}`}>
                        <a>
                            <i className="material-icons">{icon}</i>
                            <span>{name}</span>
                        </a>
                        { subItems ?
                            <ul>
                                {subItems.map((itemsname, j) => (
                                    <li key={`sidebar-subItems-${itemsname}-${j}`}>
                                        <span>{itemsname}</span>
                                    </li>
                                ))}
                            </ul>
                            : ""}
                    </li>
                ))}
            </ul>
        </nav>
    );
};

export default Sidebar;

// {subItems.map((itemsname, j) => {
//     <li key={`sidebar-itemstwo-${itemsname}-${j}`}>
//         {console.log(itemsname)}
//         <span>{itemsname}</span>
//     </li>;
// })}
