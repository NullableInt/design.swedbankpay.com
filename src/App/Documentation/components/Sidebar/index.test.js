import React from "react";
import { shallow } from "enzyme";

import Sidebar, { StandardSidebar, TwoLevels, IconsOnly, Responsive, SidebarText } from "./index";

describe("Component: Sidebar", () => {
    it("is defined", () => {
        expect(Sidebar).toBeDefined();
    });

    it("renders", () => {
        const wrapper = shallow(<Sidebar />);

        expect(wrapper).toMatchSnapshot();
    });

    describe("StandardSidebar", () => {
        it("is defined", () => {
            expect(StandardSidebar).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<StandardSidebar />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("TwoLevels", () => {
        it("is defined", () => {
            expect(TwoLevels).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<TwoLevels />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("IconsOnly", () => {
        it("is defined", () => {
            expect(IconsOnly).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<IconsOnly />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("Responsive", () => {
        it("is defined", () => {
            expect(Responsive).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<Responsive />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("SidebarText", () => {
        it("is defined", () => {
            expect(SidebarText).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<SidebarText />);

            expect(wrapper).toMatchSnapshot();
        });
    });
});
