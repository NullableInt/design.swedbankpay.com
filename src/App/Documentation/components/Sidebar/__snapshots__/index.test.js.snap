// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Sidebar Responsive renders 1`] = `
<Fragment>
  <h2
    id="responsive-sidebar"
  >
    Responsive sidebar
  </h2>
  <p>
    To only use responsive mode use class 
    <code
      className="token property"
    >
      .sidebar
    </code>
     alone.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <Sidebar
      items={
        Array [
          Object {
            "icon": "home",
            "name": "Home",
          },
          Object {
            "icon": "shopping_cart",
            "name": "Transactions",
          },
          Object {
            "icon": "favorite",
            "name": "Favorites",
          },
          Object {
            "icon": "language",
            "name": "Language",
          },
          Object {
            "icon": "account_balance",
            "name": "Account",
          },
        ]
      }
    />
  </ComponentPreview>
</Fragment>
`;

exports[`Component: Sidebar SidebarText renders 1`] = `
<div
  className="col-md-12 col-lg-10 doc-body"
>
  <p
    className="lead"
  >
    Sidebars...
  </p>
  <StandardSidebar />
  <TwoLevels />
  <TabletMode />
  <Responsive />
</div>
`;

exports[`Component: Sidebar StandardSidebar renders 1`] = `
<Fragment>
  <h2
    id="sidebar-standard"
  >
    Standard sidebar
  </h2>
  <p>
    Add class 
    <code
      className="token property"
    >
      .sidebar
    </code>
     and 
    <code
      className="token property"
    >
      .sidebar-vertical-wide-desired_size
    </code>
     to a nav containing an 
    <PrismCode
      className="language-html"
      component="code"
    >
      &lt;ul&gt;
    </PrismCode>
     with 
    <PrismCode
      className="language-html"
      component="code"
    >
      &lt;li&gt;
    </PrismCode>
     items to get a standard sidebar. Add 
    <PrismCode
      className="language-html"
      component="code"
    >
      &lt;i&gt;
    </PrismCode>
     and 
    <PrismCode
      className="language-html"
      component="code"
    >
      &lt;span&gt;
    </PrismCode>
     with your desired 
    <Link
      replace={false}
      to="/docs/core/icons"
    >
      icon
    </Link>
     and text. 
    <code
      className="token property"
    >
      .sidebar-vertical-wide-desired_size
    </code>
     determines when your sidebar will switch from responsive mode to vertical. Only the first four icons will be visible in responsive mode.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <Sidebar
      items={
        Array [
          Object {
            "icon": "home",
            "name": "Home",
          },
          Object {
            "icon": "shopping_cart",
            "name": "Transactions",
          },
          Object {
            "icon": "favorite",
            "name": "Favorites",
          },
          Object {
            "icon": "language",
            "name": "Language",
          },
          Object {
            "icon": "account_balance",
            "name": "Account",
          },
        ]
      }
      mode="wide-lg"
    />
  </ComponentPreview>
</Fragment>
`;

exports[`Component: Sidebar TabletMode renders 1`] = `
<Fragment>
  <h2
    id="tablet-sidebar"
  >
    Tablet mode
  </h2>
  <p>
    To display only icons use class 
    <code
      className="token property"
    >
      .sidebar-vertical-desired_size
    </code>
    .  
    <code
      className="token property"
    >
      .sidebar-vertical-desired_size
    </code>
     works just like in the standard example. In this mode it is possible to display another level of items.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <Sidebar
      items={
        Array [
          Object {
            "icon": "home",
            "name": "Home",
          },
          Object {
            "icon": "shopping_cart",
            "name": "Transactions",
          },
          Object {
            "icon": "favorite",
            "name": "Favorites",
            "subItems": Array [
              "Music",
              "Movies",
              "Employees",
            ],
          },
          Object {
            "icon": "language",
            "name": "Language",
          },
          Object {
            "icon": "account_balance",
            "name": "Account",
            "subItems": Array [
              "Email",
              "Information",
              "Other things",
            ],
          },
        ]
      }
      mode="md"
    />
  </ComponentPreview>
</Fragment>
`;

exports[`Component: Sidebar TwoLevels renders 1`] = `
<Fragment>
  <h2
    id="twolevels-sidebar"
  >
    Standard sidebar with two levels
  </h2>
  <p>
    You can also nest another level of items in the standard sidebar. Just nest a 
    <PrismCode
      className="language-html"
      component="code"
    >
      &lt;div&gt;
    </PrismCode>
     with class 
    <code
      className="token property"
    >
      .submenu
    </code>
     containing an icon and text like the standard sidebar and add a new 
    <PrismCode
      className="language-html"
      component="code"
    >
      &lt;ul&gt;
    </PrismCode>
     with 
    <PrismCode
      className="language-html"
      component="code"
    >
      &lt;li&gt;
    </PrismCode>
     items. This level does not use icons. List items containing another level are not visible in responsive mode, they will be displayed when the items are expanded.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <Sidebar
      items={
        Array [
          Object {
            "icon": "home",
            "name": "Home",
          },
          Object {
            "icon": "shopping_cart",
            "name": "Transactions",
          },
          Object {
            "icon": "favorite",
            "name": "Favorites",
            "subItems": Array [
              "Music",
              "Movies",
              "Employees",
            ],
          },
          Object {
            "icon": "language",
            "name": "Language",
          },
          Object {
            "icon": "account_balance",
            "name": "Account",
            "subItems": Array [
              "Email",
              "Information",
              "Other things",
            ],
          },
        ]
      }
      mode="wide-lg"
    />
  </ComponentPreview>
</Fragment>
`;

exports[`Component: Sidebar renders 1`] = `
<div
  className="doc-container"
>
  <div
    className="row"
  >
    <SidebarText />
    <DocToc
      component={[Function]}
    />
  </div>
</div>
`;
